### DISCUSSION TITLE
How does imagemagick's processing of  .DNG files into lossless .PNG's compare to that of other software like darktable/rawtherapee , quality wise ?

### DISCUSSION THREAD

**Discussion by uczpre** [ImageMagick/ImageMagick#7407] [Digital Image Processing] [answered]
i do mind the small details , so please do let me know such a difference ;D !

**Comment by fmw42** [comment]
We have not compared as far as I know.  Are you talking about quality or speed?  Please always give more details and be sure your questions are clear. We just code to the specs.  A developer might comment further.

**Comment by uczpre** [comment]
oh sorry , im talking about quality.

**Comment by snibgo** [comment, marked as answer]
IM uses libraw to read and process the raw camera file. I would expect IM to give the same quality as any other software that uses libraw, when used with the same parameters.

Libraw has not kept up with recent interpolation ("de-mosaicing") methods. Different methods have different trade-offs. As far as I can see, there is not a "highest quality" method.

See:
https://www.libraw.org/docs/API-datastruct-eng.html
https://rawpedia.rawtherapee.com/Demosaicing
https://docs.darktable.org/usermanual/3.8/en/module-reference/processing-modules/demosaic/

**Comment by uczpre** [comment]
Oh cool ! thanks a lot man ;D !
cant wait to free up 5% of my hard drive space by wasting hours of my time setting it up and encoding it !

**Reply by fmw42** [reply]
You could just use libraw. You don't need Imagemagick if that is all you want to do.
